prov/psm/src/psmx_init.c

```diff
@@ -398,7 +399,7 @@ PSM_INI
 	int check_version;
 	int err;
 
-	fi_log_init();
+	psm_handle = fi_log_register_provider("psm");
```

Comment:
We have a list of fi_prov structures in fabric.c.  I would try to just use that structure and extend it as needed, with no additional calls needed by the provider.  I'm not sure how to handle dynamic providers ,or if we even want to try to.